bootstrap/service.go

```diff
@@ -158,8 +165,14 @@ func (bs bootstrapService) Add(ctx context.Context, token string, cfg Config) (C
 		return Config{}, errors.Wrap(errThingNotFound, err)
 	}
 
+	for _, channel := range cfg.Channels {
+		if channel.DomainID != mgThing.DomainID {
+			return Config{}, errors.Wrap(svcerr.ErrMalformedEntity, errNotInSameDomain)
```

Comment:
I could not add config  and getting error , Because thing domain id is empty. 
This happens because in thing service , the json have field name `domain` but `sdk.Thing` expects field with name `domain_id`
To fix this please change here to `domain_id` https://github.com/absmach/magistrala/blob/main/pkg/clients/clients.go#L45
Or update the sdk.thing domain field json tag to `domain`